### PR TITLE
send emails on failed deploys

### PR DESCRIPTION
**Pull request by grosser**
@pswadi-zendesk @jonmoter @steved555 @princemaple @sandlerr 

screen shot 2015-02-24 at 5 53 59 pm

### CODE REVIEW

**Comment by grosser** [comment]
bump good to go, had to do some nasty hacks in the schema + added tests for them since I was running into mysql errors https://travis-ci.org/zendesk/samson/jobs/52194828

File: app/models/deploy_service.rb [hunk 1]
```diff
@@ -78,9 +78,15 @@ def send_after_notifications(stage, deploy)
     send_datadog_notification(stage, deploy)
     send_github_notification(stage, deploy)
     send_zendesk_notification(stage, deploy)
+    send_failed_deploy_notification(stage, deploy)
```
Comment (sandlerr) [inline app/models/deploy_service.rb]: `send_failed_deploy_email` ?

File: app/models/stage.rb [hunk 2]
```diff
@@ -154,6 +154,27 @@ def send_github_notifications?
...
+    return if !email_committers_on_automated_deploy_failure? && static_emails_on_automated_deploy_failure.blank?
+    return unless deploy.failed?
+    return unless deploy.user.integration?
+    last_deploy = deploys.finished_naturally.prior_to(deploy).first
+    return if last_deploy.try(:failed?)
+
+    emails = []
+
+    # static notification
+    emails.concat static_emails_on_automated_deploy_failure.to_s.split(/, ?/)
```
Comment (sandlerr) [inline app/models/stage.rb]: wouldn't the regex be `/[, ?/` ?

File: app/models/stage.rb [hunk 3]
```diff
@@ -154,6 +154,27 @@ def send_github_notifications?
...
+
+    # authors of commits after last successful deploy
+    if email_committers_on_automated_deploy_failure?
+      changeset = deploy.changeset_to(last_deploy)
+      emails.concat changeset.commits.map(&:author_email).compact
+    end
+
+    emails.presence
```
Comment (sandlerr) [inline app/models/stage.rb]: is it worth putting `emails.uniq.presence`?

**Comment by sandlerr** [comment]
:+1: assuming the non-nullable columns will work with existing records

File: app/models/stage.rb (hunk 3 above)
Comment (grosser) [inline app/models/stage.rb]: I don't think emails are getting sent in duplicates, but also no harm in doing it :)

**Pull request merged**